aten/src/ATen/native/TensorShape.cpp

```diff
@@ -896,6 +896,24 @@ Tensor numpy_T(const Tensor &self) {
   }
   return self.permute(transpose_dims);
 }
-}
 
+Tensor view(const Tensor& self, IntArrayRef size) {
+  auto inferred_size = at::infer_size(size, self.numel());
+  if (self.sizes() == inferred_size) {
+    return self;
+  }
+
+  auto stride = at::detail::computeStride(self.sizes(),
+                                           self.strides(),
+                                           inferred_size);
+  TORCH_CHECK(stride.has_value(), "view size is "
+    "not compatible with input tensor's size and stride (at least one dimension"
+    " spans across two contiguous subspaces). Use .reshape(...) instead.");
+  auto stride_value = *stride;
+  auto self_ = self.clone();
```

Comment:
wait, we are cloning in view?!?!?!

Reply:
can we _please_ do our best to have a single implementation in TH and ATen?  You can call ATen from TH, so there's usually no blocker to having a single implementation (use `THTensor_wrap` -- the only issue you might run into is extra refcounting in an inner loop, but that's pretty unlikely for `view`).

But if you had a single implementation (i.e. moved the existing one) it would be obvious that an extra `clone` call was added.

Reply:
you mean refactor TH code to call ATen function?

Reply:
Sorry, my bad - I missed the clone() in review.

Re sharing - my comment was about exactly that - redirecting to the same implementation, i.e. either using _th_view everywhere or (better) removing _th_view all together and just calling ATen function in THTensor_(newView)

Reply:
I checked codebase, looks like there is no more reference to _th_view in aten, so I'm using at::view in TH code and removing _th_view code in my fix diff.